app/src/main/java/org/eyeseetea/malariacare/layout/adapters/survey/DynamicTabAdapter.java

```diff
@@ -146,12 +144,21 @@ public DynamicTabAdapter(Tab tab, Context context) {
         this.navigationController = initNavigationController(tab);
         this.readOnly = Session.getSurvey() != null && !Session.getSurvey().isInProgress();
         this.isSwipeAdded=false;
+        //On create dynamictabadapter, if is not readonly and has value not null it should come from reviewFragment
+        if(!readOnly){
+            Question question=navigationController.getCurrentQuestion();
+            Value value = question.getValueBySession();
```

Comment:
@idelcano Something minor but there is no need to assign question.getValueBySession() to value. You can just do `if (question.getValueBySession()!=null)`